src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -1957,7 +1957,7 @@ class G1RebuildRemSetTask: public WorkerTask {
       HeapWord* const top_at_mark_start = hr->prev_top_at_mark_start();
 
       HeapWord* cur = hr->bottom();
-      while (cur < hr->end()) {
+      while (cur < hr->top()) {
```

Comment:
I think it is possible (and less misleading) to just use a `while (true) {` loop. Neither a condition containing `end()` or `top()` is correct. The only boundary that matters is `top_at_rebuild_start()`; since it can change between iterations, and it is needed in the loop I do not see a good condition here.
(Using `top()` may exit the loop just a tiny bit earlier than the `next_chunk.is_empty()` check below though, but for simplicity, i.e. having one exit condition less to think about, it might be useful to just state `true` here).

Reply:
Agree, will upate the patch.